### DISCUSSION TITLE
Updating SuperStream Connection Access Token

### DISCUSSION THREAD

**Discussion by lngvivek** [rabbitmq/rabbitmq-stream-dotnet-client#334] [Q&A]
We are using SuperStream dotnet client version 1.7.2
For Authentication with Rabbit, we are using IdenitityServer Acces Token. It works perfectly on initial connection. However once the access toke expires, we were looking for something UpdateSecret (as used to be in Queue based sdk) on Connection object so that our connection is updated with lastest Access token. Could not find any such option for updating token to RabbitMQ.Stream.Client.StreamSystem. Could anyone pls suggest on how to refresh the token?

**Comment by Gsantomaggio** [comment]
Thank you for reporting the issue.
There is not at the moment. We have to add it. It is on the list. 

Feel free to open a PR.

**Reply by lngvivek** [reply]
@Gsantomaggio Would be great if you could let me know any tentative timelines when we can expect this enhancement.

**Reply by Gsantomaggio** [reply]
@lngvivek will do after: https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/discussions/333#discussioncomment-7850816 

Thank you

**Comment by Gsantomaggio** [comment]
@lngvivek FYI: https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/pull/342

**Reply by vivek-singh2** [reply]
@Gsantomaggio Thank you very much.
